NoSoft-SA/nspack
# Feature/titan duplicate pallets with multiple fields

Summary of this change:
After some investigation, it was found that TITAN expects the following fields on the pallet header level instead of pallet sequence:
 - Grade/Inspection Class 
 - Inspection-/Marketing Variety
 - PHC
To account for this odd data structure, pallets with multiple values for the above-mentioned fields are duplicated and regrouped by each unique set of pallet sequences with these fields and sent to TITAN accordingly.

CHANGELOG entry:

### Added
- `includes_duplicated_pallets` to `titan_requests`.
- `duplicate_pallets` to `TitanRepo` which is called if a pallet comprises multiple values for `class, variety or phc`.
### Changed
- Refactored functions `compile_inspection`, `compile_inspection_pallets` and `compile_inspection_pallet_sequences` in`TitanRepo` to support this change. 

How to test this code via the UI:
Try to TITAN a pallet with multiple values for the above-mentioned field. 
Kromco pallet  `460023035009891314` was used, for example.
Verify that normal pallets still work as well.
